core/src/main/scala/cats/data/EitherT.scala

```diff
@@ -88,11 +88,22 @@ final case class EitherT[F[_], A, B](value: F[Either[A, B]]) {
 
   def map[D](f: B => D)(implicit F: Functor[F]): EitherT[F, A, D] = bimap(identity, f)
 
+  def mapF[N[_], D](fe: F[Either[A, B]] => N[Either[A, D]]): EitherT[N, A, D] = EitherT(fe(value))
```

Comment:
@kailuowang taken your advice I opened #1713 and came to the conclusion to add this to EitherT

```scala
def transformF[G[_]](f: FunctionK[F, G]): EitherT[G, A, B] = EitherT(f(value))
```

There is something confusing about EitherT and OptionT using `transform` as a bimap/map of the parametrized types respectively. I called it transformF just to avoid the confusion. The issue is opened for the evaluation of your proposal and to address the confusion around `transform` semantics.